test/platform/test-platform-identify.ts

```diff
@@ -0,0 +1,131 @@
+      await t.throwsAsync<Error>(
+        mockLocalHomePlatform.triggerIdentify(DISCOVERY_BUFFER)
+      )
+    ).message,
+    ERROR_UNDEFINED_APP
+  );
+});
+
+test('trigger-identify-without-listen-throws', async t => {
+  const mockLocalHomePlatform = MockLocalHomePlatform.getInstance(true);
+  const app: AppStub = new AppStub(APP_VERSION);
+  t.is(
+    (
+      await t.throwsAsync<Error>(
+        mockLocalHomePlatform.triggerIdentify(DISCOVERY_BUFFER)
+      )
+    ).message,
+    ERROR_LISTEN_NOT_CALLED
+  );
+});
+
+test('trigger-identify-with-undefined-handler-throws', async t => {
+  const mockLocalHomePlatform = MockLocalHomePlatform.getInstance(true);
+  const app: AppStub = new AppStub(APP_VERSION);
+  app.listen();
+  t.is(
+    (
+      await t.throwsAsync<Error>(
+        mockLocalHomePlatform.triggerIdentify(DISCOVERY_BUFFER)
+      )
+    ).message,
+    ERROR_UNDEFINED_IDENTIFYHANDLER
+  );
+});
+
+test('trigger-identify-with-undefined-verificationId-throws', async t => {
+  const mockLocalHomePlatform = MockLocalHomePlatform.getInstance(true);
+  const app: AppStub = new AppStub(APP_VERSION);
+  const invalidIdentifyHandler: smarthome.IntentFlow.IdentifyHandler = () => {
+    return {
+      requestId: 'request-id',
+      intent: smarthome.Intents.IDENTIFY,
+      payload: {
+        device: {
+          id: DEVICE_ID,
+        },
+      },
+    };
+  };
+  app.onIdentify(invalidIdentifyHandler).listen();
+  t.is(
+    (
+      await t.throwsAsync<Error>(
+        mockLocalHomePlatform.triggerIdentify(DISCOVERY_BUFFER)
+      )
+    ).message,
+    ERROR_UNDEFINED_VERIFICATIONID
+  );
+});
+
+test('trigger-identify-with-valid-state', async t => {
+  const discoveryBuffer = Buffer.from('discovery buffer 123');
+  const mockLocalHomePlatform = MockLocalHomePlatform.getInstance(true);
+  const localDeviceId = 'local-device-id-123';
+  const validIdentifyHandler: smarthome.IntentFlow.IdentifyHandler = () => {
+    return {
+      requestId: 'request-id',
+      intent: smarthome.Intents.IDENTIFY,
+      payload: {
+        device: {
+          id: DEVICE_ID,
+          verificationId: localDeviceId,
+        },
+      },
+    };
+  };
+  const app: AppStub = new AppStub(APP_VERSION);
+  app.onIdentify(validIdentifyHandler).listen();
+
+  t.notThrows(() => {
+    mockLocalHomePlatform.triggerIdentify(discoveryBuffer);
+  });
+  t.is(await mockLocalHomePlatform.getNextDeviceIdRegistered(), localDeviceId);
+  t.is(mockLocalHomePlatform.getLocalDeviceIdMap().size, 1);
+  t.is(
+    mockLocalHomePlatform.getLocalDeviceIdMap().keys().next().value,
+    DEVICE_ID
+  );
+  t.is(
+    mockLocalHomePlatform.getLocalDeviceIdMap().values().next().value,
+    localDeviceId
+  );
+});
+
+test('trigger-identify-without-reset-state-throws', async t => {
+  const oldPlatform = MockLocalHomePlatform.getInstance(true);
+  const app: AppStub = new AppStub(APP_VERSION);
+  const mockLocalHomePlatform = MockLocalHomePlatform.getInstance(false);
```

Comment:
Do we need to get this twice?

Reply:
This tests the behavior of the `getInstance()` reset state flag.  The first `getInstance()` call is necessary so the difference between the two flag values can be compared.  I replaced this test with the opposite case since this failing case is kind of arbitrary.  The new test initializes a `MockLocalHomePlatform`, sets an `App`, then creates a new `MockLocalHomePlatform` with `resetState = true`, and assets that there is no `App` set in the new `MockLocalHomePlatform`